### PR TITLE
Add address entity, associated it with customer

### PR DESCRIPTION
**Pull request by suiyoubi**

### CODE REVIEW

File: src/main/java/com/ubc/cpsc304/hotelify/controller/CustomerController.java [hunk 1]
```diff
@@ -42,11 +51,62 @@ private CustomerResponseDto createCustomer(@RequestBody CustomerRequestDto custo
...
+    @GetMapping("/{username}")
+    private CustomerResponseDto getCustomer(@PathVariable String username)
+            throws BadRequestException {
+
+        Customer customer = this.customerService.findByUsername(username);
+
+        return CustomerController.convertModel(customer);
+    }
+
+    @PostMapping("/{username}/address")
```
Comment (chenyirex) [inline src/main/java/com/ubc/cpsc304/hotelify/controller/CustomerController.java]: I actually like `("/address/{username}"` better

File: src/main/java/com/ubc/cpsc304/hotelify/service/CustomerService.java [hunk 2]
```diff
@@ -19,6 +22,17 @@ public CustomerService(CustomerRepository customerRepository) {
...
     }
 
+    public Customer findByUsername(String username) throws BadRequestException {
+
+        Optional<Customer> optionalCustomer = this.customerRepository.findById(username);
+
+        if (optionalCustomer.isPresent()) {
+            return optionalCustomer.get();
+        } else {
+            throw new BadRequestException("username does not exist");
```
Comment (chenyirex) [inline src/main/java/com/ubc/cpsc304/hotelify/service/CustomerService.java]: should be a 404 when you cant find the resource queried, I think.

File: src/main/java/com/ubc/cpsc304/hotelify/controller/CustomerController.java (hunk 1 above)
Comment (suiyoubi) [reply src/main/java/com/ubc/cpsc304/hotelify/controller/CustomerController.java]: In my understand:
user **has** address. so the address is under user URL

File: src/main/java/com/ubc/cpsc304/hotelify/service/CustomerService.java (hunk 2 above)
Comment (suiyoubi) [reply src/main/java/com/ubc/cpsc304/hotelify/service/CustomerService.java]: 👍

File: src/main/java/com/ubc/cpsc304/hotelify/controller/CustomerController.java (hunk 1 above)
Comment (chenyirex) [reply src/main/java/com/ubc/cpsc304/hotelify/controller/CustomerController.java]: But it is POST /customers/{username}/address. 
So I think address is under customers, and {username} should be used to specify which user is it for this operation.

File: src/main/java/com/ubc/cpsc304/hotelify/controller/CustomerController.java [hunk 3]
```diff
@@ -42,11 +51,62 @@ private CustomerResponseDto createCustomer(@RequestBody CustomerRequestDto custo
...
+    private AddressDto addAddressInformationToCustomer(
+            @PathVariable String username,
+            @RequestBody AddressDto addressDto
+    ) throws BadRequestException, ConflictException {
+
+        Customer customer = this.customerService.findByUsername(username);
+
+        if (Objects.nonNull(customer.getAddress())) {
+            throw new ConflictException("customer already have an address, use PUT for change");
```
Comment (chenyirex) [inline src/main/java/com/ubc/cpsc304/hotelify/controller/CustomerController.java]: capitialized **C** and **has**

**Review by chenyirex — APPROVED** [review]

**Pull request merged**